Formula/llvm.rb

```diff
@@ -109,8 +93,8 @@ class Llvm < Formula
   uses_from_macos "zlib"
 
   patch :p1 do
-    url "https://raw.githubusercontent.com/Homebrew/formula-patches/c7da61787c62822804dd90192dd9dd254511192c/llvm/10.0.1-llvm.diff"
-    sha256 "86b4e2bf10dfb9a04153e3d4fc744c3f323f3e13fc7b3a7d22e6791d47eb2e2c"
+    url "https://raw.githubusercontent.com/kubkon/formula-patches/llvm-11-system-libs-patch/llvm/11.0.0-llvm.diff"
```

Comment:
@jedisct1 @fxcoudert a temp to see if the patch works. If it does, I'll go ahead and submit a PR to [Homebrew/formula-patches](https://github.com/Homebrew/formula-patches).

Reply:
```text
==> Applying 11.0.0-llvm.diff
can't find file to patch at input line 5
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/llvm/lib/WindowsManifest/CMakeLists.txt b/llvm/lib/WindowsManifest/CMakeLists.txt
|index fe6ddcd414d5..8cfea1282ad6 100644
|--- a/llvm/lib/WindowsManifest/CMakeLists.txt
|+++ b/llvm/lib/WindowsManifest/CMakeLists.txt
--------------------------
No file to patch.  Skipping patch.
```

I'm not even sure that this is the right patch though. All these discussions and unrelated patches are quite confusing.

Reply:
The path should be `llvm-11.0.0.src/lib/WindowsManifest/CMakeLists.txt`.

The patch should also probably be reverted, not applied:

```diff
--- llvm-11.0.0.src/lib/WindowsManifest/CMakeLists.txt	2020-10-07 12:10:48.000000000 +0200
+++ llvm-11.0.0.src/lib/WindowsManifest/CMakeLists.txt	2020-10-13 21:16:16.000000000 +0200
@@ -8,16 +8,6 @@
 if(LIBXML2_LIBRARIES)
   target_link_libraries(LLVMWindowsManifest PUBLIC ${LIBXML2_LIBRARIES})
 
-  get_filename_component(xml2_library ${LIBXML2_LIBRARIES} NAME)
-  if (CMAKE_STATIC_LIBRARY_PREFIX AND
-      xml2_library MATCHES "^${CMAKE_STATIC_LIBRARY_PREFIX}.*${CMAKE_STATIC_LIBRARY_SUFFIX}$")
-    string(REGEX REPLACE "^${CMAKE_STATIC_LIBRARY_PREFIX}" "" xml2_library ${xml2_library})
-    string(REGEX REPLACE "${CMAKE_STATIC_LIBRARY_SUFFIX}$" "" xml2_library ${xml2_library})
-  elseif (CMAKE_SHARED_LIBRARY_PREFIX AND
-          xml2_library MATCHES "^${CMAKE_SHARED_LIBRARY_PREFIX}.*${CMAKE_SHARED_LIBRARY_SUFFIX}$")
-    string(REGEX REPLACE "^${CMAKE_SHARED_LIBRARY_PREFIX}" "" xml2_library ${xml2_library})
-    string(REGEX REPLACE "${CMAKE_SHARED_LIBRARY_SUFFIX}$" "" xml2_library ${xml2_library})
-  endif()
   set_property(TARGET LLVMWindowsManifest PROPERTY
-    LLVM_SYSTEM_LIBS ${xml2_library})
+    LLVM_SYSTEM_LIBS ${LIBXML2_LIBRARIES})
 endif()
```

I started a compile with it. Let's see how it goes.

Reply:
It works!

And fixes the issue we had with Zig.

Reply:
Awesome! Thanks for taking the time to verify this and for all your help here @jedisct1!

Reply:
It actually picks up the same `tbd` file as before, but with the full path and without the extra `-l`.

Do we really want `-lxml2` instead? We can hardcode it, but this patch should restore the behavior from LLVM 10.0.0 that worked well.

Reply:
No, I'd rather modify the added test case than force `-lxml2`. So with this patch, we are actually getting a fully qualified path to `libxml2.tbd`, did I get it right?

Reply:
Correct.

Reply:
I've now tweaked the test to assert for this situation plus a situation where `llvm-config --system-libs` reports a fully qualified path to a `dylib` instead of `tbd` (as evident on my installation, for some reason).